SummaryNumbers: close on click for mobile

psealock opened the pull request:
Fixes https://github.com/woocommerce/wc-admin/issues/1096

Close Summary Numbers menu when selecting a summary number on smaller viewports.

SummaryList now accepts a function as children instead of an array of Summary Numbers. The function allows the passage of the `onToggle` parameter provided by Gutenberg's `<Dropdown />`.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)

### Detailed test instructions:

1. Dashboard > Store Performance.
2. Ensure Summary Numbers render correctly.
3. Revenue Report Ensure Summary Numbers render correctly at large viewports.
4. Check again on smaller viewports.
5. When selecting a chart type (Net Revenue, for example) the dropdown should close automatically.

psealock committed:
SummaryNumbers: close on click for mobile

psealock commented on client/dashboard/store-performance/index.js:
```diff
@@ -77,33 +77,46 @@ class StorePerformance extends Component {
+								);
+							}
+							return summaryNumbers;
+						} }
```
This is ugly, but its getting redone in https://github.com/woocommerce/wc-admin/pull/1343 anyways

psealock commented on packages/components/src/summary/number.js:
```diff
@@ -144,6 +146,10 @@ SummaryNumber.propTypes = {
+	/**
+	 * A function to be called after a SummaryNumber, rendered as a link, is clicked.
+	 */
+	onLinkCallback: PropTypes.func,
```
I'm open to naming suggestions here. `onToggle` is used to determine if component is a link or button and I'd like to provide a way for a function to be called after the link has been activated.

joshuatf commented on packages/components/src/summary/number.js:
```diff
@@ -60,6 +61,7 @@ const SummaryNumber = ( {
 	if ( ! onToggle ) {
```
Just for my clarification, this is called only when `onToggle` is passed from a dropdown when this is a button, right?

Optional, but what do you think about creating a constant `isButton` instead of checking `onToggle`?  Took me a second to figure out what was going on here and this might make it a tiny bit faster to understand.

joshuatf reviewed (commented):
This is testing well.  Few minor semantic comments, but otherwise looks ready to merge.

psealock commented:
Thanks @joshuatf !

* Created a `getHasItemsClass` util.
* Removed unnecessary defaultProp.
* Renamed `onLinkClickCallback`.
* Added an `isButton` constant for clarity.

jeffstieler commented on packages/components/src/summary/index.js:
```diff
@@ -3,59 +3,44 @@
-		label = __( 'Performance Indicators', 'wc-admin' );
-	}
-
+	const items = children( {} );
```
Optional nitpick:

psealock replied:
The child function of `report-summary/index.js` does however. I'll get `Cannot read property 'onToggle' of undefined` if I remove this argument.

jeffstieler reviewed (approved):
Code looks good and tests well - one optional nitpick change. 🚢 it!
